tests/zfs-tests/tests/functional/fault/suspend_on_probe_errors.ksh

```diff
@@ -0,0 +1,151 @@
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2024, Klara Inc.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/include/blkdev.shlib
+
+#
+# DESCRIPTION: Verify that 4 disks removed from a raidz3 will suspend the pool
+#
+# STRATEGY:
+# 1. Disable ZED -- this test is focused on vdev_probe errors
+# 2. Create a raidz3 pool where 4 disks can be removed (i.e., using scsi_debug)
+# 3. Add some data to it for a resilver workload
+# 4. Replace one of the child vdevs to start a replacing vdev
+# 5. During the resilver, remove 4 disks including one from the replacing vdev
+# 6. Verify that the pool is suspended (it used to remain online)
+#
+
+DEV_SIZE_MB=1024
+
+FILE_VDEV_CNT=8
+FILE_VDEV_SIZ=256M
+
+function cleanup
+{
+	destroy_pool $TESTPOOL
+	if [[ "$(cat /sys/block/$sd/device/state)" == "offline" ]]; then
+		log_must eval "echo running > /sys/block/$sd/device/state"
+	fi
+	unload_scsi_debug
+	rm -f $DATA_FILE
+	for i in {0..$((FILE_VDEV_CNT - 1))}; do
+		log_must rm -f "$TEST_BASE_DIR/dev-$i"
+	done
+	zed_start
+}
+
+log_onexit cleanup
+
+log_assert "VDEV probe errors for more disks than parity should suspend a pool"
+
+log_note "Stoping ZED process"
+zed_stop
+zpool events -c
+
+# Make a debug device that we can "unplug" and lose 4 drives at once
+unload_scsi_debug
+load_scsi_debug $DEV_SIZE_MB 1 1 1 '512b'
+sd=$(get_debug_device)
+
+# Create 4 partitions that match the FILE_VDEV_SIZ
+parted "/dev/${sd}" --script mklabel gpt
+parted "/dev/${sd}" --script mkpart primary 0% 25%
+parted "/dev/${sd}" --script mkpart primary 25% 50%
+parted "/dev/${sd}" --script mkpart primary 50% 75%
+parted "/dev/${sd}" --script mkpart primary 75% 100%
+block_device_wait "/dev/${sd}"
+blkdevs="/dev/${sd}1 /dev/${sd}2 /dev/${sd}3 /dev/${sd}4"
+
+# Create 8 file vdevs
+typeset -a filedevs
+for i in {0..$((FILE_VDEV_CNT - 1))}; do
+	device=$TEST_BASE_DIR/dev-$i
+	log_must truncate -s $FILE_VDEV_SIZ $device
+	# Use all but the last one for pool create
+	if [[ $i -lt "7" ]]; then
+		filedevs[${#filedevs[*]}+1]=$device
+	fi
+done
+
+# Create a raidz-3 pool that we can pull 4 disks from
+log_must zpool create -f $TESTPOOL raidz3 ${filedevs[@]} $blkdevs
+sync_pool $TESTPOOL
+
+# Add some data to the pool
+log_must zfs create $TESTPOOL/fs
+MNTPOINT="$(get_prop mountpoint $TESTPOOL/fs)"
+SECONDS=0
+log_must fill_fs $MNTPOINT 3 200 32768 300 Z
```

Comment:
I see one of the ZTS runs has `fill_fs` taking 2 minutes:
```
21:46:11.39 SUCCESS: fill_fs /testpool/fs 3 200 32768 300 Z                      
21:46:11.39 NOTE: fill_fs took 121.807 seconds 
```
I reduced the values and tested on my local VM and was still able to pass:
```diff
- log_must fill_fs $MNTPOINT 3 200 32768 300 Z
+ log_must fill_fs $MNTPOINT 1 200 32768 1 Z
```
It could maybe be reduced further but I didn't go down the rabbit hole.

Reply:
I have suspicion that this test might be time-sensitive, and reducing this size too much may even more increase the chance of scrub completion before the device is offlined.

Reply:
> I see one of the ZTS runs has `fill_fs` taking 2 minutes:
> 
> ```
> 21:46:11.39 SUCCESS: fill_fs /testpool/fs 3 200 32768 300 Z                      
> 21:46:11.39 NOTE: fill_fs took 121.807 seconds 
> ```

@tonyhutter  was there something unique about the VM for that run?  Do you happen to know if there is a  minimum system that we should consider when writing tests?  This tests needs enough data in the pool for the resilver to still be in-flight when the disks are pulled (i.e. a resilver that takes at least a few seconds to complete).  I can retest for the minimum system we think will be used.